### PR TITLE
BOD 23-02 customer report changes

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR:
* Adds a new section in the CyHy customer report related to [BOD 23-02](https://www.cisa.gov/news-events/directives/binding-operational-directive-23-02)
* Makes some minor changes to the "Report Card" section of the report
* Adds a new "possible_nmi" field to the `potentially-risky-services.csv` attachment

## 💭 Motivation and context ##

This PR contains the changes requested in https://github.com/cisagov/cyhy-system/issues/89.

## 🧪 Testing ##

I built a local Docker image with these changes and then used that to generate a variety of test reports.  I ensured that the reports looked as expected (for both Federal Executive and non-Federal Executive organizations), in terms of the new content changes and modifications to the `potentially-risky-services.csv` attachment.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Post-merge checklist ##

- [x] Deploy this change to Production.

### CODE REVIEW

File: cyhy_report/customer/report.mustache [hunk 1]
```diff
@@ -748,7 +748,11 @@ As you review the report, you may have additional questions. Check out the answe
     }
 
     \vspace*{2mm}  % blank line needed on previous line for vertical spacing
-    \RCSourceSansProLightEight{\textit{Service counts are best guesses and may not be 100\% accurate. Details can be found in ''potentially-risky-services.csv'' in \autoref{app:attachments}.}}
+    \RCSourceSansProLightEight{\textit{Service counts are best guesses and may not be 100\% accurate. Details can be found in ''potentially-risky-services.csv'' in \autoref{app:attachments}.
```
Comment (jsf9k) [inline cyhy_report/customer/report.mustache]:
```suggestion
    \RCSourceSansProLightEight{\textit{Service counts are best guesses and may not be 100\% accurate. Details can be found in ``potentially-risky-services.csv'' in \autoref{app:attachments}.
```

File: cyhy_report/customer/report.mustache [hunk 2]
```diff
@@ -825,6 +829,26 @@ Details on the below findings can be found in ``findings.csv" in
...
+
+\gls{CISA} <<#owner_is_federal_executive>>issued \href{https://www.cisa.gov/news-events/directives/binding-operational-directive-23-02}{Binding Operational Directive (BOD) 23-02} to push the federal government to take steps toward reducing the attack surface created by insecure or misconfigured management interfaces across certain classes of devices. The BOD requires networked management interfaces (NMIs) using certain protocols over the Internet to <</owner_is_federal_executive>><<^owner_is_federal_executive>>recommends that networked management interfaces (NMIs) using certain protocols over the Internet <</owner_is_federal_executive>>be removed from the public Internet or to be protected by capabilities that enforce access control to the interface through a policy enforcement point separate from the interface itself as part of a Zero Trust Architecture (ZTA) within 14 days of discovery.
```
Comment (jsf9k) [inline cyhy_report/customer/report.mustache]:
```suggestion
\gls{CISA} <<#owner_is_federal_executive>>issued \href{https://www.cisa.gov/news-events/directives/binding-operational-directive-23-02}{Binding Operational Directive (BOD) 23-02} to push the federal government to take steps toward reducing the attack surface created by insecure or misconfigured management interfaces across certain classes of devices. The BOD requires networked management interfaces (NMIs) using certain protocols over the Internet to <</owner_is_federal_executive>><<^owner_is_federal_executive>>recommends that networked management interfaces (NMIs) using certain protocols over the Internet to <</owner_is_federal_executive>>be removed from the public Internet or to be protected by capabilities that enforce access control to the interface through a policy enforcement point separate from the interface itself as part of a Zero Trust Architecture (ZTA) within 14 days of discovery.
```

File: cyhy_report/customer/report.mustache [hunk 3]
```diff
@@ -825,6 +829,26 @@ Details on the below findings can be found in ``findings.csv" in
...
+
+We also recommend reviewing all hosts with potentially risky open services, especially if they are functioning as networked management interfaces, to ensure that each service is intended to be available to the public and, where applicable, the service is up to date on the latest version, correctly configured, and uses strong authentication.
+
+You can find a list of potentially risky services detected as available on your external network within this report's ``potentially-risky-services.csv" attachment. In it, there is a column which denotes those that may be associated with NMIs to help with prioritization.
```
Comment (jsf9k) [inline cyhy_report/customer/report.mustache:843]:
```suggestion
You can find a list of potentially risky services detected as available on your external network within this report's ``potentially-risky-services.csv'' attachment. In it, there is a column which denotes those that may be associated with NMIs to help with prioritization.
```

File: cyhy_report/customer/report.mustache [hunk 4]
```diff
@@ -825,6 +829,26 @@ Details on the below findings can be found in ``findings.csv" in
...
+
+\begin{figure}[H]
+	\centering
+	\scalegraphics{potential-nmi-service-counts}{Potential Network Management Interface (NMI) Service Counts}
+\end{figure}
+
+The details for these findings can be found within the ``potentially-risky-services.csv" file located in \autoref{app:attachments}: \nameref{app:attachments}. You will need to ensure you open the report with a dedicated PDF reader (such as Adobe Acrobat), and click on the paperclip icon to the left of the CSV file in order to open it.
```
Comment (jsf9k) [inline cyhy_report/customer/report.mustache:850]:
```suggestion
The details for these findings can be found within the ``potentially-risky-services.csv'' file located in \autoref{app:attachments}: \nameref{app:attachments}. You will need to ensure you open the report with a dedicated PDF reader (such as Adobe Acrobat), and click on the paperclip icon to the left of the CSV file in order to open it.
```

**Review by jsf9k — APPROVED** [review]
I made some suggestions, which you can take or leave.  Also, I'm not sure if `"` renders the same as `''` in LaTeX.  I have always used the latter.

File: cyhy_report/customer/report.mustache (hunk 2 above)
Comment (dav3r) [reply cyhy_report/customer/report.mustache]: I don't think adding the word "to" here makes sense.  In any event, that is the language that I was given in https://github.com/cisagov/cyhy-system/issues/89:

> CISA recommends that networked management interfaces (NMIs) using certain protocols over the Internet be removed from the public Internet...

File: cyhy_report/customer/report.mustache (hunk 3 above)
Comment (dav3r) [reply cyhy_report/customer/report.mustache:843]: I don't plan to change this.  This document has always used this style of double-quote LaTeX formatting:
```
``quoted words"
```

File: cyhy_report/customer/report.mustache (hunk 4 above)
Comment (dav3r) [reply cyhy_report/customer/report.mustache:850]: I don't plan to change this.  This document has always used this style of double-quote LaTeX formatting:
```
``quoted words"
```

Comment (jsf9k) [reply cyhy_report/customer/report.mustache:850]: OK, I guess LaTeX treats `"` identically to `''` then.

File: cyhy_report/customer/report.mustache (hunk 3 above)
Comment (jsf9k) [reply cyhy_report/customer/report.mustache:843]: OK, I guess LaTeX treats `"` identically to `''` then.

File: cyhy_report/customer/report.mustache (hunk 2 above)
Comment (jsf9k) [reply cyhy_report/customer/report.mustache]: The problem I was trying to fix is that with non-Feds you currently end up with:
```
CISA recommends that [NMIs] be removed from the public Internet or TO be protected by...
```

That extra "to" is problematic.

Comment (jsf9k) [reply cyhy_report/customer/report.mustache]: How about this?
```
\gls{CISA} <<#owner_is_federal_executive>>issued \href{https://www.cisa.gov/news-events/directives/binding-operational-directive-23-02}{Binding Operational Directive (BOD) 23-02} to push the federal government to take steps toward reducing the attack surface created by insecure or misconfigured management interfaces across certain classes of devices. The BOD requires networked management interfaces (NMIs) using certain protocols over the Internet to <</owner_is_federal_executive>><<^owner_is_federal_executive>>recommends networked management interfaces (NMIs) using certain protocols over the Internet to <</owner_is_federal_executive>>be removed from the public Internet or to be protected by capabilities that enforce access control to the interface through a policy enforcement point separate from the interface itself as part of a Zero Trust Architecture (ZTA) within 14 days of discovery.
```

Comment (jsf9k) [reply cyhy_report/customer/report.mustache]: Aargh.  I'll do another review with the change as a suggestion for clarity.

Comment (jsf9k) [inline cyhy_report/customer/report.mustache]:
```suggestion
\gls{CISA} <<#owner_is_federal_executive>>issued \href{https://www.cisa.gov/news-events/directives/binding-operational-directive-23-02}{Binding Operational Directive (BOD) 23-02} to push the federal government to take steps toward reducing the attack surface created by insecure or misconfigured management interfaces across certain classes of devices. The BOD requires networked management interfaces (NMIs) using certain protocols over the Internet to <</owner_is_federal_executive>><<^owner_is_federal_executive>>recommends networked management interfaces (NMIs) using certain protocols over the Internet to <</owner_is_federal_executive>>be removed from the public Internet or to be protected by capabilities that enforce access control to the interface through a policy enforcement point separate from the interface itself as part of a Zero Trust Architecture (ZTA) within 14 days of discovery.
```

**Review by jsf9k — COMMENTED** [review]
Here is a better suggestion for this block of text.

File: cyhy_report/customer/report.mustache (hunk 2 above)
Comment (jsf9k) [reply cyhy_report/customer/report.mustache]: Please see [here](https://github.com/cisagov/cyhy-reports/pull/89#pullrequestreview-1552649168) instead.

Comment (dav3r) [reply cyhy_report/customer/report.mustache]: As I was looking through your suggestion, I realized that my output differed slightly from the text provided in the [requirements](https://github.com/cisagov/cyhy-system/issues/89).  I made corrections in https://github.com/cisagov/cyhy-reports/pull/89/commits/16a1f76fd9135570fa29369ba180b534f5fc5ade.

If you still have an issue with this, then I think you might have a disagreement with the text provided in the requirements.

Comment (jsf9k) [reply cyhy_report/customer/report.mustache]: That commit looks good to me.  Thank you!

**Review by jmorrowomni — APPROVED** [review]
LGTM!

**Review by jasonodoom — APPROVED** [review]

**Pull request merged**